scripts/fgi/kube.py

```diff
@@ -0,0 +1,258 @@
+# Copyright (c) The Diem Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import json, random, subprocess, time
+
+FORGE_K8S_CLUSTERS = ["forge-1"]
+
+# ================ Kube job ================
+def create_forge_job(context, user, tag, timeout_secs, forge_envs, forge_args):
+    job_name = f"forge-{user}-{int(time.time())}"
+    job_name = job_name.replace("_", "-")  # underscore not allowed in pod name
+
+    # job template to spin up. Edit this in place
+    template = json.loads(
+        subprocess.check_output(
+            [
+                "kubectl",
+                "-o=json",
+                f"--context={context}",
+                "get",
+                "job",
+                "--selector=app.kubernetes.io/name=forge-debug",
+            ],
+            stderr=subprocess.DEVNULL,
+            encoding="UTF-8",
+        )
+    )
+    if len(template["items"]) != 1:
+        print("ERROR: there must be exactly one forge-debug job")
+        return None
+
+    template = template["items"][0]
+
+    # delete old spec details
+    del template["metadata"]["selfLink"]
+    del template["spec"]["selector"]["matchLabels"]["controller-uid"]
+    del template["spec"]["template"]["metadata"]["labels"]["controller-uid"]
+    del template["spec"]["template"]["metadata"]["labels"]["job-name"]
+    # change job name, labels, and backoff limit
+    template["metadata"]["name"] = job_name
+    template["metadata"]["labels"]["app.kubernetes.io/name"] = "forge"
+    template["spec"]["template"]["metadata"]["labels"][
+        "app.kubernetes.io/name"
+    ] = "forge"
+    template["spec"]["backoffLimit"] = 0
+    # change startup command with timeout and extra args
+    cmd = template["spec"]["template"]["spec"]["containers"][0]["command"][2]
+    template["spec"]["template"]["spec"]["containers"][0]["command"][2] = cmd.replace(
+        "tail -f /dev/null",
+        f"timeout {timeout_secs} forge {' '.join(forge_args)}".strip(),
+    )
+    # additional environment variables
+    for env_var in forge_envs:
+        name, value = env_var.split("=")
+        template["spec"]["template"]["spec"]["containers"][0]["env"].append(
+            {"name": name, "value": value}
+        )
+    # new image tag
+    image_repo, _ = template["spec"]["template"]["spec"]["containers"][0][
+        "image"
+    ].split(":")
+    template["spec"]["template"]["spec"]["containers"][0][
+        "image"
+    ] = f"{image_repo}:{tag}"
+    return job_name, template
+
+
+# ================ Kube queries ================
+
+
+def get_cluster_context(cluster_name):
+    k8s_context_pattern = "arn:aws:eks:us-west-2:853397791086:cluster/libra-CLUSTERNAME"
```

Comment:
I think we did hide account number here in fgi?

Reply:
updated to get it via the caller identity